### PR TITLE
Add optional requestHeaders argument to autoUpdater.setFeedURL

### PR DESCRIPTION
**Pull request by miniak**
This allows setting custom HTTP request headers, such as Authorization.

### CODE REVIEW

**Comment by zcbenz** [comment]
How about making the headers a parameter of `setFeedURL`? Like this:

```js
```

**Comment by miniak** [comment]
@zcbenz that's what I wanted to do at first, but I couldn't find an example in the code, how to get a key/value pair object input argument. Can you help me with that? I know how to make the headers argument optional.

**Comment by ssova-4xxi** [comment]
@miniak what about windows?

**Comment by miniak** [comment]
Squirrel for Windows does not seem to have an option to set HTTP headers

**Comment by ssova-4xxi** [comment]
@miniak maybe add warning under windows?

**Comment by miniak** [comment]
In the docs?

File: lib/browser/api/auto-updater/auto-updater-win.js [hunk 1]
```diff
@@ -23,6 +23,9 @@ AutoUpdater.prototype.setFeedURL = function (updateURL) {
   this.updateURL = updateURL
 }
 
+AutoUpdater.prototype.setRequestHeader = function (header, value) {
```
Comment (ssova-4xxi) [inline lib/browser/api/auto-updater/auto-updater-win.js]: @miniak here

**Comment by miniak** [comment]
Is stating that in the documentation not enough? I have to check what Electron does in other methods, which are platform specific

**Comment by zcbenz** [comment]
> @zcbenz that's what I wanted to do at first, but I couldn't find an example in the code, how to get a key/value pair object input argument. Can you help me with that? I know how to make the headers argument optional.

You need to first converting the optional argument into a `base::DictionaryValue`, and then iterate it. An example of iterating the `base::DictionaryValue` can be found at:
https://github.com/electron/electron/blob/master/atom/browser/net/atom_network_delegate.cc#L161-L174

**Comment by miniak** [comment]
I didn't have time to work on this during the week, I am going to finish it over the weekend.

**Comment by miniak** [comment]
@zcbenz can you please check again?

**Comment by zcbenz** [comment]
👍

**Pull request merged**